gtda/externals/bindings/collapser_bindings.cpp

```diff
@@ -72,7 +72,7 @@ PYBIND11_MODULE(gtda_collapser, m) {
           for (size_t k = 0; k < size; ++k)
             for (Eigen::SparseMatrix<Filtration_value>::InnerIterator it(sm, k);
                  it; ++it) {
-              if (it.value() <= thresh)
+              if (it.col() > it.row() && it.value() <= thresh)
```

Comment:
Maybe worth commenting, for our future selves, that unlike the `ripser` function collapser is going to ignore lower-diagonal entries?

Reply:
What do you mean by `collapser is going to ignore` ?
The lower part *must* be ignored, otherwise the computation can go wrong.

Would you prefer a comment about this directly into the code?

Reply:
I should have said "our collapser bindings are going to ignore". What I mean to say is that the user of `ripser` (and `VietorisRipsPersistence`) is told that it's OK to pass sparse distance matrices with lower triangular entries, because (as of #537) we don't ignore them (unless entries also exist in the corresponding transpose positions). But the power users of the collapser binding functions (if there are any) should know that this is not the case if they use collapser in isolation.

Reply:
One might also *think* that the GUDHI implementation takes care of this (but we now know it does not!).

Reply:
> But the power users of the collapser binding functions (if there are any)
Let me believe there are thousands of them, please

Then, I  should document the code, but in my opinion, if the power user want to use `collapser` in isolation, we should write a documentation, or is there an existing one ?

Reply:
Well, maybe in the case of `gudhi`, because as far as I understood, it used inside of the `simplex_tree`, they already meet this requirement, I don't know

Reply:
So here my suggestion reduces to just replacing lines 70 and 100 with
```
          /* Applying threshold to the input data and ignoring lower diagonal entries */
```
What do you think?